### PR TITLE
docs(masthead): storybook demo cleanup

### PR DESCRIPTION
**Pull request by emyarod**
### Related Ticket(s)

#6407

### Description

This PR updates all of the masthead stories according to the changes outlined in #6407.

For the "With custom navigation" story, should we just remove it instead of hiding it? Given that it is a storybook-only change I don't think it would be a breaking change even though the ticket states otherwise, right?

### CODE REVIEW

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 1]
```diff
@@ -200,16 +205,14 @@ export const withPlatform = ({ parameters }) => {
           ></dds-masthead-composite>
         `
       : html`
-          <dds-masthead-container
-            platform="${ifNonNull(platformData.name)}"
+          <dds-masthead-composite
```
Comment (IgnacioBecerra) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:208]: Just wondering the reason behind the change here?

**Review by IgnacioBecerra — COMMENTED** [review]
Overall looks fine, I also think removing the story would be better since it's not a breaking change.

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts (hunk 1 above)
Comment (emyarod) [reply packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:208]: this was to fulfill the "Remove all links from With platform story other than the Platform" requirement in the original issue. From what I could tell, the container will always render the links https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/master/packages/web-components/src/components/masthead/masthead-container.ts#L86

**Review by kennylam — APPROVED** [review]
LGTM thanks @emyarod!

**Comment by oliviaflory** [comment]
Hey @emyarod 
I saw your note about deleting the `With custom navigation` story, makes sense that it's not a breaking change being only in storybook! 

This looks really good, I noticed two things:

1. There's a bit of an odd thing happening with the recent update to the Global nav label from Services & Consulting to Consulting & Services. The Consulting & Services only shows as selected if you type "Services & Consulting", but not if you type "Consulting & Services". 
 
I'm not 100% sure why that is, but guessing it has something to do with that label originally being "Services & Consulting"
Screen Shot 2021-12-07 at 3 48 41 PM
Screen Shot 2021-12-07 at 3 48 56 PM

2. I missed requesting to hide the Platform knob on the `With alternate logo and tooltip` story. Is it difficult to remove that too?
Screen Shot 2021-12-07 at 4 18 04 PM

I saw Shixie asked in the original issue if updating the e2e tests would be part of this, let us know if we need a separate issue to cover that 🙂

**Comment by emyarod** [comment]
opened https://github.ibm.com/webstandards/ibm-dotcom-translations/pull/544 to address the selected menu highlight mismatch

**Comment by emyarod** [comment]
@oliviaflory the platform knob should now be gone from that story, and the issue with the highlighted/selected menu will be resolved in the previous PR I linked. I will update the e2e tests in this PR once all of the visual changes are confirmed to be ok!

**Review by oliviaflory — COMMENTED** [review]
Looks good @emyarod, thank you!

There are a few unexpected visual difs in the React e2e Percy test, but I don't believe anything in this PR would have caused that.

**Review by annawen1 — APPROVED** [review]
LGTM!

**Comment by annawen1** [comment]
@emyarod looks like the e2e tests need to be adjusted, can you take a look?

**Comment by ariellalgilmore** [comment]
Looks like it's still failing now for some reason on the masthead default selected menu item test. It passes though for the L1...

**Comment by emyarod** [comment]
yeah that's really strange, even when I view the cypress storybook locally the menu item is properly highlighted for both of the stories with and without L1. I'm not sure why it fails when running headlessly though

**Comment by emyarod** [comment]
I'm not able to approve the percy builds but the changes look good to me

**Pull request merged**